perceval/utils/postselect.py

```diff
@@ -26,206 +26,190 @@
-                  "<=": int.__le__}
-
-    # Regexp explanations:
-    # first group: index(es) of modes between '[]' and separated with ','
-    # second group: operator (listed in _OPERATORS). We suppose there is no digit character in operators
-    # third group: number of photons
-    _PATTERN = re.compile(r"(\[[,0-9\s]+\]\s*)([^\d]*)\s*(\d+\b)")
-
-    def __init__(self, str_repr: str = None):
-        self._conditions = {}
-        condition_count = 0
-        if str_repr:
-            try:
-                for match in self._PATTERN.finditer(str_repr):
-                    indexes = tuple(json.loads(match.group(1)))
-
-                    operator = match.group(2).strip()
-                    if operator not in self._OPERATORS:
-                        raise KeyError(f"Unsupported operator: {operator}")
-
-                    self._add_condition(indexes=indexes,
-                                        operator=self._OPERATORS[operator],
-                                        value=int(match.group(3)))
-                    condition_count += 1
-
-            except json.decoder.JSONDecodeError as e:
-                raise RuntimeError(f"Could not interpret input string '{str_repr}': {e}")
-            if condition_count != str_repr.count("&") + 1:
-                raise RuntimeError(f"Could not interpret input string '{str_repr}': Invalid format")
-
-    def eq(self, indexes, value: int):
-        """Create a new "equals"     condition for the current PostSelect instance"""
-        self._add_condition(indexes, int.__eq__, value)
-        return self
-
-    def gt(self, indexes, value: int):
-        """Create a new "greater than" condition for the current PostSelect instance"""
-        self._add_condition(indexes, int.__gt__, value)
-        return self
-
-    def lt(self, indexes, value: int):
-        """Create a new "lower than" condition for the current PostSelect instance"""
-        self._add_condition(indexes, int.__lt__, value)
-        return self
-
-    def ge(self, indexes, value: int):
-        """Create a new "greater or equal than" condition for the current PostSelect instance"""
-        self._add_condition(indexes, int.__ge__, value)
-        return self
-
-    def le(self, indexes, value: int):
-        """Create a new "lower or equal than" condition for the current PostSelect instance"""
-        self._add_condition(indexes, int.__le__, value)
-        return self
-
-    def _add_condition(self, indexes, operator: callable, value: int):
-        indexes = (indexes,) if isinstance(indexes, int) else tuple(indexes)
-        if operator not in self._conditions:
-            self._conditions[operator] = []
-        self._conditions[operator].append((indexes, value))
+    >>> ps = PostSelect("([0,1] == 1 & [2] > 1) | [2] == 0") # Means "I want either exactly one photon in mode 0 or 1, and at least one photon in mode 2, either no photon in mode 2"
+    >>> print(ps(BasicState([0, 1, 1])))
+    True
```

Comment:
Isn't this one False?

Reply:
either is usually followed by or, so I would suggest # Means "I want either exactly one photon in mode 0 or 1, and at least one photon in mode 2, **or** no photon in mode 2"